evalml/pipelines/components/transformers/imputers/imputer.py

```diff
@@ -162,6 +162,7 @@ def transform(self, X, y=None):
             X_numeric = X.ww[self._numeric_cols.tolist()]
             imputed = self._numeric_imputer.transform(X_numeric)
             X_no_all_null[X_numeric.columns] = imputed
+            X_no_all_null = downcast_integer_nullable_to_double(X_no_all_null)
```

Comment:
For more info: the error I was getting was that there was a logical type mismatch with the IntegerNullable columns when calling `downcast_boolean_nullable_to_double()`. Not too sure about the reasoning but my guess was that there was a mismatch in the IntegerNullable and the imputed numeric columns.

Reply:
I think it makes sense to treat them the same way. If we did, could we set the imputed integer columns to Integer?

Reply:
@fjlanasa sounds good. It has to be a float because we can impute by the `mean` of a column. However, I could set it to `Integer` when we use  "median", "most_frequent", "constant" but I wanted the type to be consistent coming out of the imputer. Do you think I should change it to `Integer` based off of the imputation strategy?

Reply:
Ah ok. Whatever you think makes sense.